messaging/src/test/java/org/axonframework/common/infra/JacksonComponentDescriptorTest.java

```diff
@@ -0,0 +1,461 @@
+                                    "stringValue":"test",
+                                    "numberValue":42,
+                                    "booleanValue":true,
+                                    "component":{
+                                        "_id":"*",
+                                        "_type":"SimpleTestComponent",
+                                        "name":"nestedComponent",
+                                        "value":300
+                                    },
+                                    "simpleList":["listItem1","listItem2"],
+                                    "componentList":[
+                                        {
+                                            "_id":"*",
+                                            "_type":"SimpleTestComponent",
+                                            "name":"listComponent1",
+                                            "value":301
+                                        },
+                                        {
+                                            "_id":"*",
+                                            "_type":"SimpleTestComponent",
+                                            "name":"listComponent2",
+                                            "value":302
+                                        }
+                                    ],
+                                    "simpleMap":{"mapKey1":"mapValue1"}
+                                }
+                            }
+                            """
+            );
+        }
+    }
+
+    @Disabled("TODO #XXXX - support circular references")
+    @Nested
+    class CircularReferencesTests {
+
+        @Test
+        void describeComponentWithCircularReference() {
+            // given
+            var component1 = new CircularComponent("Component1");
+            var component2 = new CircularComponent("Component2");
+            component1.setDependency(component2);
+            component2.setDependency(component1);
+            testSubject.describeProperty("circularRef", component1);
+
+            // when
+            var result = testSubject.describe();
+
+            // then
+            assertJsonMatches(result, """
+                    {
+                      "circularRef": {
+                        "_id": "%s",
+                        "_type": "CircularComponent",
+                        "name": "Component1",
+                        "dependency": {
+                          "_id": "%s",
+                          "_type": "CircularComponent",
+                          "name": "Component2",
+                          "reference": {
+                            "$ref": "%s"
+                          }
+                        }
+                      }
+                    }
+                    """.formatted(identityOf(component1), identityOf(component2), identityOf(component1))
+            );
+        }
```

Comment:
This will be addressed in following PRs. The idea is similar like in MongoDB, to use `$ref` and the `id`, which is the object identity to reference if there is a cycle.